flink-connectors/flink-parquet/src/main/java/org/apache/flink/api/io/parquet/ParquetSchemaConverter.java

```diff
@@ -0,0 +1,167 @@
+	 */
+	private TypeInformation<?> convertType(Type parquetType) {
+		if (parquetType.isPrimitive()) {
+			return convertPrimitiveType(parquetType.asPrimitiveType());
+		} else {
+			return convertGroupType(parquetType.asGroupType());
+		}
+	}
+
+	/**
+	 * Converts a primitive Parquet {@link Type} to a Flink {@link TypeInformation}.
+	 */
+	private TypeInformation<?> convertPrimitiveType(PrimitiveType primitiveType) {
+		PrimitiveType.PrimitiveTypeName typeName = primitiveType.getPrimitiveTypeName();
+		OriginalType originalType = primitiveType.getOriginalType();
+
+		switch (typeName) {
+			case BOOLEAN:
+				return BasicTypeInfo.BOOLEAN_TYPE_INFO;
+			case FLOAT:
+				return BasicTypeInfo.FLOAT_TYPE_INFO;
+			case DOUBLE:
+				return BasicTypeInfo.DOUBLE_TYPE_INFO;
+			case INT32:
+				return convertOriginalType_INT32(originalType);
+			case INT64:
+				return convertOriginalType_INT64(originalType);
+			case BINARY:
+				return convertOriginalType_BINARY(originalType);
+			case INT96:
+			case FIXED_LEN_BYTE_ARRAY:
+			default:
+				throw new UnsupportedOperationException(typeName + " is not supported");
+		}
+	}
+
+	private TypeInformation<?> convertOriginalType_INT32(OriginalType originalType) {
+		if (originalType == null) {
+			return BasicTypeInfo.INT_TYPE_INFO;
+		}
+		switch (originalType) {
+			case INT_8:
+				return BasicTypeInfo.BYTE_TYPE_INFO;
+			case INT_16:
+				return BasicTypeInfo.SHORT_TYPE_INFO;
+			case INT_32:
+				return BasicTypeInfo.INT_TYPE_INFO;
+			case DATE:
+				return BasicTypeInfo.DATE_TYPE_INFO;
+			case DECIMAL:
+				return BasicTypeInfo.BIG_DEC_TYPE_INFO;
+			case UINT_8:
+			case UINT_16:
+			case UINT_32:
+			case TIME_MILLIS:
+			default:
+				throw new UnsupportedOperationException(originalType + " is not supported");
+		}
+	}
+
+	private TypeInformation<?> convertOriginalType_INT64(OriginalType originalType) {
+		if (originalType == null) {
+			return BasicTypeInfo.LONG_TYPE_INFO;
+		}
+		switch (originalType) {
+			case INT_64:
+				return BasicTypeInfo.LONG_TYPE_INFO;
+			case DECIMAL:
+				return BasicTypeInfo.BIG_DEC_TYPE_INFO;
+			case TIMESTAMP_MILLIS:
+			case UINT_64:
+			default:
+				throw new UnsupportedOperationException(originalType + " is not supported");
+		}
+	}
+
+	private TypeInformation<?> convertOriginalType_BINARY(OriginalType originalType) {
+		if (originalType == null) {
+			return PrimitiveArrayTypeInfo.BYTE_PRIMITIVE_ARRAY_TYPE_INFO;
+		}
+		switch (originalType) {
+			case UTF8:
+			case ENUM:
+			case JSON:
+				return BasicTypeInfo.STRING_TYPE_INFO;
+			case BSON:
+				return PrimitiveArrayTypeInfo.BYTE_PRIMITIVE_ARRAY_TYPE_INFO;
+			case DECIMAL:
+				return BasicTypeInfo.BIG_DEC_TYPE_INFO;
+			default:
+				throw new UnsupportedOperationException(originalType + " is not supported");
+		}
+	}
+
+	/**
+	 * Converts a combined Parquet {@link Type} to a Flink {@link TypeInformation}.
+	 */
+	private TypeInformation<?> convertGroupType(GroupType groupType) {
```

Comment:
Without implementing GroupType, map and array can't be handled. Is there any plan to add it also?